bionic/protocols.py

```diff
@@ -186,8 +185,16 @@ def __call__(self, func=None, **kwargs):
                     )
                 )
 
-            wrapper = decorator_wrapping_provider(ProtocolUpdateProvider, self)
-            return wrapper(func)
+            def update_accumulator(acc):
```

Comment:
You're right! I lost track of the fact that we weren't checking this before. I agree that it's unlikely anyone is doing this but it's good to avoid breakage on principle. I'll change this to a warning.